index_test.go

```diff
@@ -240,7 +239,184 @@ func approxSame(actual, expected uint64) bool {
 
 	return float64(modulus(actual, expected))/float64(expected) < float64(0.25)
 }
+func TestBytesWritten(t *testing.T) {
+	tmpIndexPath := createTmpIndexPath(t)
+	defer cleanupTmpIndexPath(t, tmpIndexPath)
+
+	indexMapping := NewIndexMapping()
+	indexMapping.TypeField = "type"
+	indexMapping.DefaultAnalyzer = "en"
+	documentMapping := NewDocumentMapping()
+	indexMapping.AddDocumentMapping("hotel", documentMapping)
+
+	indexMapping.DocValuesDynamic = false
+	indexMapping.StoreDynamic = false
+
+	contentFieldMapping := NewTextFieldMapping()
+	contentFieldMapping.Index = true
+	contentFieldMapping.Store = false
+	contentFieldMapping.IncludeInAll = false
+	contentFieldMapping.IncludeTermVectors = false
+	contentFieldMapping.DocValues = false
+
+	reviewsMapping := NewDocumentMapping()
+	reviewsMapping.AddFieldMappingsAt("content", contentFieldMapping)
+	documentMapping.AddSubDocumentMapping("reviews", reviewsMapping)
+
+	typeFieldMapping := NewTextFieldMapping()
+	typeFieldMapping.Store = false
+	typeFieldMapping.IncludeInAll = false
+	typeFieldMapping.IncludeTermVectors = false
+	typeFieldMapping.DocValues = false
+	documentMapping.AddFieldMappingsAt("type", typeFieldMapping)
+	idx, err := NewUsing(tmpIndexPath, indexMapping, Config.DefaultIndexType, Config.DefaultMemKVStore, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var wg sync.WaitGroup
+	batch, err := getBatchFromData(idx, "sample-data.json")
+	if err != nil {
+		t.Fatalf("failed to form a batch %v\n", err)
+	}
+	wg.Add(1)
+	batch.SetPersistedCallback(func(e error) {
+		defer wg.Done()
+		stats, _ := idx.StatsMap()["index"].(map[string]interface{})
+		bytesWritten, _ := stats["num_bytes_written_at_index_time"].(uint64)
+		if bytesWritten != 37767 {
+			t.Fatalf("expected bytes written is 37767, got %v",
+				bytesWritten)
+		}
+	})
+	err = idx.Batch(batch)
+	if err != nil {
+		t.Fatalf("failed to index batch %v\n", err)
+	}
+	wg.Wait()
+	idx.Close()
+
+	contentFieldMapping.Store = true
+	tmpIndexPath1 := createTmpIndexPath(t)
+	defer cleanupTmpIndexPath(t, tmpIndexPath1)
+
+	idx, err = NewUsing(tmpIndexPath1, indexMapping, Config.DefaultIndexType, Config.DefaultMemKVStore, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	batch, err = getBatchFromData(idx, "sample-data.json")
+	if err != nil {
+		t.Fatalf("failed to form a batch %v\n", err)
+	}
+	wg.Add(1)
+	batch.SetPersistedCallback(func(e error) {
+		defer wg.Done()
+		stats, _ := idx.StatsMap()["index"].(map[string]interface{})
+		bytesWritten, _ := stats["num_bytes_written_at_index_time"].(uint64)
+		if bytesWritten != 56582 {
+			t.Fatalf("expected bytes written is 56582, got %v",
+				bytesWritten)
+		}
+	})
+	err = idx.Batch(batch)
+	if err != nil {
+		t.Fatalf("failed to index batch %v\n", err)
+	}
+	wg.Wait()
+	idx.Close()
+
+	contentFieldMapping.IncludeInAll = true
+	tmpIndexPath2 := createTmpIndexPath(t)
+
```

Comment:
clean up of this path is missing?
Also, should we need to defer it as we could try it right after the index close? (doesn't matter much)
It could also be written like a table-driven test since the variations across run are only a few values.